### PR TITLE
Balance Twelve-based layout for Niu Mini

### PR DESCRIPTION
**Pull request by yttyx**
## Description

Addition of a Balance Twelve-based layout for the Niu Mini, which is aimed at left-handers who use a mouse with their right hand (e.g. me!).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [X ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/niu_mini/keymaps/yttyx/config.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#ifndef CONFIG_H
```
Comment (mechmerlin) [inline keyboards/niu_mini/keymaps/yttyx/config.h]: You can replace lines 1 and 2 with just `#pragma once`

File: keyboards/niu_mini/keymaps/yttyx/keymap.c [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  BA,           // Base        (Balance Twelve mirror varian)
+  P1,           // Punctuation (1)
+  P2,           // Punctuation (2)
+  P3,           // Punctuation (2)
+  NC,           // Numerals / Cursor control
+  FV            // Function keys / Cursor control (Vim)
+};
+
+// Abbreviations - general
+#define xxxxxxx         KC_NO
```
Comment (mechmerlin) [inline keyboards/niu_mini/keymaps/yttyx/keymap.c]: if you use capital Xs, this is already included in `#include QMK_KEYBOARD_H`

```suggestion

```

File: keyboards/niu_mini/keymaps/yttyx/keymap.c [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+#define _______         KC_TRNS
```
Comment (mechmerlin) [inline keyboards/niu_mini/keymaps/yttyx/keymap.c]: This is already included in `#include QMK_KEYBOARD_H`
```suggestion

```

File: keyboards/niu_mini/keymaps/yttyx/config.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+// Disable some options to reduce firmware size
+#define NO_ACTION_TAPPING
+#define NO_ACTION_ONESHOT
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define NO_DEBUG
+#define NO_PRINT
+
+#endif
```
Comment (noroadsleft) [inline keyboards/niu_mini/keymaps/yttyx/config.h]: And when you replace Lines 1 and 2, delete this line at the same time.

```suggestion
```

File: keyboards/niu_mini/keymaps/yttyx/keymap.c (hunk 3 above)
Comment (yttyx) [reply keyboards/niu_mini/keymaps/yttyx/keymap.c]: Fixed.

File: keyboards/niu_mini/keymaps/yttyx/config.h (hunk 4 above)
Comment (yttyx) [reply keyboards/niu_mini/keymaps/yttyx/config.h]: Making this change created an issue whereby no layer transitions occurred. Please see general comment for more information.

**Comment by yttyx** [comment]
Thank you both for your comments.

The proposed changes to config.h caused issues. I have listed the variants of config.h that I tried, together with the results, below. I would welcome your advice on how to proceed! It seems like very odd behaviour.

------------------------------------------------
WORKS
```

#ifndef CONFIG_H
#define CONFIG_H

// Disable some options to reduce firmware size
#define NO_ACTION_TAPPING
#define NO_ACTION_ONESHOT
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
#define NO_DEBUG
#define NO_PRINT

#endif

```
------------------------------------------------
WORKS
```

#ifndef CONFIG_H
#define CONFIG_H

// Disable some options to reduce firmware size
#define NO_ACTION_TAPPING
#define NO_ACTION_ONESHOT
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
//#define NO_DEBUG
//#define NO_PRINT
```

#endif

------------------------------------------------
Builds, but layer transitions do not work
```

#pragma once

// Disable some options to reduce firmware size
#define NO_ACTION_TAPPING
#define NO_ACTION_ONESHOT
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
//#define NO_DEBUG
//#define NO_PRINT
```

------------------------------------------------
Build errors: 'NO_DEBUG and NO_PRINT redefined'
```

#pragma once

// Disable some options to reduce firmware size
#define NO_ACTION_TAPPING
#define NO_ACTION_ONESHOT
#define NO_ACTION_MACRO
#define NO_ACTION_FUNCTION
#define NO_DEBUG
#define NO_PRINT
```
------------------------------------------------

File: keyboards/niu_mini/keymaps/yttyx/keymap.c (hunk 2 above)
Comment (yttyx) [reply keyboards/niu_mini/keymaps/yttyx/keymap.c]: Fixed.

File: keyboards/niu_mini/keymaps/yttyx/config.h (hunk 1 above)
Comment (yttyx) [reply keyboards/niu_mini/keymaps/yttyx/config.h]: Making this change created an issue whereby no layer transitions occurred. Please see general comment for more information.

Comment (drashna) [reply keyboards/niu_mini/keymaps/yttyx/config.h]: That may be because you have `NO_ACTION_TAPPING` defined.

Comment (yttyx) [reply keyboards/niu_mini/keymaps/yttyx/config.h]: Hi drashna. Many thanks!  I made the check, and as you suggest, `CONFIG_H` _is_ already defined. I have amended `config.h` accordingly and successfully retested. The change is pushed.

**Comment by yttyx** [comment]
I believe the requested changes have all been made now.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**